tests/python/unittest/test_tir_schedule_transform_layout.py

```diff
@@ -533,6 +533,21 @@ def before():
     expected = tvm.tir.schedule.schedule.ScheduleError
 
 
+class TestErrorOnNonMatchingTypes(BasePaddingCompare):
+    """The padding must have the same dtype as the buffer"""
+
+    pad_value = tvm.testing.parameter(0)
+
+    def before():
+        A = T.alloc_buffer(14, "float32")
+        for i in T.serial(14):
+            with T.block("block"):
+                vi = T.axis.remap("S", [i])
+                A[vi] = 0
+
+    expected = TypeError
```

Comment:
Does the behavior here differ from [TestErrorOnWrongPaddingType](https://github.com/apache/tvm/blob/main/tests/python/unittest/test_tir_schedule_transform_layout.py#L521)?  Other than the type of error thrown, it looks like they are testing the same behavior.

Reply:
Ah, got it.  A test of usage error where an entirely incompatible type is being passed, rather than just a test that requires a different Python to TIR conversion.  Thank you!